_source/_posts/2020-01-08-oauth-legacy-spring-cloud-gateway.adoc

```diff
@@ -0,0 +1,299 @@
+---
+layout: blog_post
+title: "Add 2.0 OAuth to Legacy Apps with Spring Cloud Gateway"
+author: bdemers
+description: "This tutorial shows show you how to secure legacy applications with OAuth 2.0 and Spring Cloud Gateway"
+tags: [java, spring cloud gateway, legacy, spring boot, spring cloud, oauth, oidc]
+tweets:
+- "A low code option to secure legacy applications with Spring Cloud Gateway!"
+- "Adding OAuth 2.0 support to your existing applications is easy with Spring Cloud Gateway"
+- "Legacy Apps + Spring Cloud Gateway == 🔒"
+image: blog/oauth-legacy-spring-cloud-gateway/spring-cloud-gateway-legacy-social.png
+---
+:page-liquid:
+
+One of the biggest challenges of adding OAuth 2.0 support to legacy applications is a lack of support in the underlying framework. Maybe it's homegrown, or maybe it's just old? Either way, migrating away from an old form-based login doesn't need to be so painful. In this post, I'll walk you through a low-code option using Spring Cloud Gateway and Okta.
+
+You'll learn how to setup Spring Cloud Gateway running as a stand-alone application that proxies HTTP requests and handles the OAuth dance before forwarding the request onto your application.
+
+image::{% asset_path 'blog/oauth-legacy-spring-cloud-gateway/spring-cloud-gateway-legacy.png' %}[alt=Spring Cloud Gateway sequence diagram,width=800,align=center]
+
+== Create a Spring Cloud Gateway Project
+
+The first step is to create a new project on the https://start.spring.io[Spring Initializr]. I typically create it from my IDE or the command line:
+
+[source,text]
```

Comment:
```suggestion
[source,shell]
```
The reason I suggest changing this is because the code type shows up in the top right.